src/System.Runtime.Extensions/tests/System/IO/PathTests_Join.netcoreapp.cs

```diff
@@ -117,5 +117,77 @@ public void TryJoinThreePaths(string path1, string path2, string path3, string e
+            { "a", "", "", "b", $"a{Sep}b" },
+            { "a", "b", "c", "", $"a{Sep}b{Sep}c" },
+            { "a", "b", "", "c", $"a{Sep}b{Sep}c" },
+            { "a", "", "b", "c", $"a{Sep}b{Sep}c" },
+            { "", "a", "b", "c", $"a{Sep}b{Sep}c" },
+            { "a", "b", "c", "d", $"a{Sep}b{Sep}c{Sep}d" },
+            { "a", Sep, "b", "", $"a{Sep}b" },
+            { "a", Sep, "", "b", $"a{Sep}b" },
+            { "a", "", Sep, "b", $"a{Sep}b" },
+            { $"a{Sep}", "b", "", "", $"a{Sep}b" },
+            { $"a{Sep}", "", "b", "", $"a{Sep}b" },
+            { $"a{Sep}", "", "", "b", $"a{Sep}b" },
+            { "", $"a{Sep}", "b", "", $"a{Sep}b" },
+            { "", $"a{Sep}", "", "b", $"a{Sep}b" },
+            { "", "", $"a{Sep}", "b", $"a{Sep}b" },
+            { "a", $"{Sep}b", "", "", $"a{Sep}b" },
+            { "a", "", $"{Sep}b", "", $"a{Sep}b" },
+            { "a", "", "", $"{Sep}b", $"a{Sep}b" },
+            { $"a{AltSep}", "b", "", "", $"a{AltSep}b" },
+            { $"a{AltSep}", "", "b", "", $"a{AltSep}b" },
+            { $"a{AltSep}", "", "", "b", $"a{AltSep}b" },
+            { "", $"a{AltSep}", "b", "", $"a{AltSep}b" },
+            { "", $"a{AltSep}", "", "b", $"a{AltSep}b" },
+            { "", "", $"a{AltSep}", "b", $"a{AltSep}b" },
+            { "a", $"{AltSep}b", "", "", $"a{AltSep}b" },
+            { "a", "", $"{AltSep}b", "", $"a{AltSep}b" },
+            { "a", "", "", $"{AltSep}b", $"a{AltSep}b" },
+            { null, null, null, null, "" },
+            { "a", null, null, null, "a" },
+            { null, "a", null, null, "a" },
+            { null, null, "a", null, "a" },
+            { null, null, null, "a", "a" },
+            { "a", null, "b", null, $"a{Sep}b" },
+            { "a", null, null, "b", $"a{Sep}b" }
+        };
+
+        [Theory, MemberData(nameof(TestData_JoinFourPaths))]
+        public void JoinFourPaths(string path1, string path2, string path3, string path4, string expected)
+        {   
+            Assert.Equal(expected, Path.Join(path1.AsSpan(), path2.AsSpan(), path3.AsSpan(), path4.AsSpan()));
+            Assert.Equal(expected, Path.Join(path1, path2, path3, path4));
+        }
+
+        [Theory, MemberData(nameof(TestData_JoinTwoPaths))]
+        public void JoinStringArray_2(string path1, string path2, string expected)
+        {
+            Assert.Equal(expected, Path.Join(new string[] { path1, path2 }));
+        }
+
+        [Theory, MemberData(nameof(TestData_JoinThreePaths))]
+        public void JoinStringArray_3(string path1, string path2, string path3, string expected)
+        {
+            Assert.Equal(expected, Path.Join(new string[] { path1, path2, path3 }));
+        }
+
+        [Theory, MemberData(nameof(TestData_JoinFourPaths))]
+        public void JoinStringArray_4(string path1, string path2, string path3, string path4, string expected)
+        {
+            Assert.Equal(expected, Path.Join(new string[] { path1, path2, path3, path4 }));
```

Comment:
can we have test for more than 4 as well ? maybe just pass the all the paths twice